LibGit2 callback error in automerge for private registry

With an automerge.yaml set up nearly identically to that used in General, we've been getting the following consistently on our private registry:
```
    Cloning registry from "https://github.com/JuliaRegistries/General.git"
25l    Fetching: [>                                        ]  0.0 %
    Fetching: [=================>                       ]  41.9 %
    Resolving Deltas: [================================>        ]  79.9 %
      Added registry `General` to `/tmp/jl_e18Pfl/registries/General`
25hTest Summary:                                | Pass  Total
(Registry|Package|Versions|Deps|Compat).toml |  287    287
[ Info: Authenticated to GitHub as "github-actions[bot]"
[ Info: Attempting to clone...
[ Info: Attempting to clone...
ERROR: GitError(Code:EUSER, Class:Callback, Aborting, user cancelled credential request.)
Stacktrace:
 [1] macro expansion at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/LibGit2/src/error.jl:101 [inlined]
 [2] clone(::String, ::String, ::LibGit2.CloneOptions) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/LibGit2/src/repository.jl:459
 [3] clone(::String, ::String; branch::String, isbare::Bool, remote_cb::Ptr{Nothing}, credentials::Nothing, callbacks::Dict{Symbol,Tuple{Ptr{Nothing},Any}}) at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/LibGit2/src/LibGit2.jl:580
 [4] clone at /buildworker/worker/package_linux64/build/usr/share/julia/stdlib/v1.4/LibGit2/src/LibGit2.jl:559 [inlined]
 [5] _clone_repo_into_dir(::String, ::String) at /home/runner/.julia/packages/RegistryCI/ZPKNe/src/AutoMerge/util.jl:25
[6] #111 at /home/runner/.julia/packages/RegistryCI/ZPKNe/src/AutoMerge/util.jl:16 [inlined]
 [7] (::Base.var"#58#60"{Base.var"#58#59#61"{ExponentialBackOff,Nothing,RegistryCI.AutoMerge.var"#111#113"{String,String}}})(; kwargs::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}) at ./error.jl:301
 [8] #58 at ./error.jl:284 [inlined]
 [9] my_retry at /home/runner/.julia/packages/RegistryCI/ZPKNe/src/AutoMerge/api_rate_limiting.jl:2 [inlined] (repeats 2 times)
 [10] clone_repo(::String) at /home/runner/.julia/packages/RegistryCI/ZPKNe/src/AutoMerge/util.jl:16
 [11] clone_repo at /home/runner/.julia/packages/RegistryCI/ZPKNe/src/AutoMerge/util.jl:10 [inlined]
 [12] pull_request_build(::GitHub.PullRequest, ::String, ::GitHub.Repo, ::String; auth::GitHub.OAuth2, authorized_authors::Array{String,1}, authorized_authors_special_jll_exceptions::Array{String,1}, master_branch::String, master_branch_is_default_branch::Bool, suggest_onepointzero::Bool, whoami::String, registry_deps::Array{String,1}) at /home/runner/.julia/packages/RegistryCI/ZPKNe/src/AutoMerge/pull-requests.jl:88
 [13] pull_request_build(::Int64, ::String, ::GitHub.Repo, ::String; whoami::String, auth::GitHub.OAuth2, authorized_authors::Array{String,1}, authorized_authors_special_jll_exceptions::Array{String,1}, master_branch::String, master_branch_is_default_branch::Bool, suggest_onepointzero::Bool, registry_deps::Array{String,1}) at /home/runner/.julia/packages/RegistryCI/ZPKNe/src/AutoMerge/pull-requests.jl:42
[14] run(::Base.EnvDict, ::RegistryCI.AutoMerge.GitHubActions; merge_new_packages::Bool, merge_new_versions::Bool, new_package_waiting_period::Minute, new_jll_package_waiting_period::Minute, new_version_waiting_period::Minute, new_jll_version_waiting_period::Minute, registry::String, authorized_authors::Array{String,1}, authorized_authors_special_jll_exceptions::Array{String,1}, additional_statuses::Array{String,1}, additional_check_runs::Array{String,1}, master_branch::String, master_branch_is_default_branch::Bool, suggest_onepointzero::Bool, registry_deps::Array{String,1}) at /home/runner/.julia/packages/RegistryCI/ZPKNe/src/AutoMerge/public.jl:50
 [15] top-level scope at none:5
Username for 'https://github.com': Username for 'https://github.com': 
##[error]Process completed with exit code 1.
```

--- a/bors.toml
+++ b/bors.toml
@@ -6,7 +6,7 @@ status = [
     # "codecov/patch",
     # "codecov/project",
     "Travis CI - Branch",
-    "VersionVigilante",
+    # "VersionVigilante",
 ]
 
 timeout_sec = 10800

--- a/src/AutoMerge/guidelines.jl
+++ b/src/AutoMerge/guidelines.jl
@@ -415,7 +415,7 @@ function _run_pkg_commands(working_directory::String,
     # process. For example, we don't want to pass an environment variable containing
     # our GitHub token to the child process. Because if the Julia package that we are
     # testing has malicious code in its __init__() function, it could try to steal
-    # our token. So we only pass eight environment variables:
+    # our token. So we only pass these environment variables:
     # 1. HTTP_PROXY. If it's set, it is delegated to the child process.
     # 2. HTTPS_PROXY. If it's set, it is delegated to the child process.
     # 3. JULIA_DEPOT_PATH. We set JULIA_DEPOT_PATH to the temporary directory that

--- a/src/AutoMerge/guidelines.jl
+++ b/src/AutoMerge/guidelines.jl
@@ -115,7 +115,7 @@ function meets_name_ascii(pkg)
 end
 
 function meets_julia_name_check(pkg)
-    if occursin("julia", lowercase(pkg)) 
+    if occursin("julia", lowercase(pkg))
         return false, "Lowercase package name $(lowercase(pkg)) contains the string \"julia\"."
     elseif startswith(pkg, "Ju")
         return false, "Package name starts with \"Ju\"."
@@ -150,7 +150,7 @@ function meets_distance_check(pkg_name, other_packages; DL_lowercase_cutoff = 1,
             if dl_lowercase <= DL_lowercase_cutoff
                 msg = string(msg, " Damerau-Levenshtein distance $(dl_lowercase) between lowercased names is at or below cutoff of $(DL_lowercase_cutoff).")
             end
-            
+
             # Distance check 3: normalized visual distance,
             # gated by a `dl` check for speed.
             if (sqrt_normalized_vd_cutoff > 0 && dl <= 4)
@@ -169,7 +169,7 @@ function meets_distance_check(pkg_name, other_packages; DL_lowercase_cutoff = 1,
             end
         end
     end
-    
+
     isempty(problem_messages) && return (true, "")
     sort!(problem_messages, by = Base.tail)
     message = string("Package name similar to $(length(problem_messages)) existing package",
@@ -429,24 +429,22 @@ function _run_pkg_commands(working_directory::String,
     #    PyCall to install their own version of Python instead of using the system
     #    Python.
     # 8. R_HOME. We set R_HOME to "*".
-    env = Dict{String, String}()
-    if haskey(ENV, "HTTP_PROXY")
-        env["HTTP_PROXY"] = deepcopy(ENV["HTTP_PROXY"])
-    end
-    if haskey(ENV, "HTTPS_PROXY")
-        env["HTTPS_PROXY"] = deepcopy(ENV["HTTPS_PROXY"])
-    end
-    env["JULIA_DEPOT_PATH"] = deepcopy(tmp_dir_2)
-    if haskey(ENV, "JULIA_PKG_SERVER")
-        env["JULIA_PKG_SERVER"] = deepcopy(ENV["JULIA_PKG_SERVER"])
-    end
-    env["JULIA_REGISTRYCI_AUTOMERGE"] = "true"
-    env["PATH"] = deepcopy(ENV["PATH"])
-    env["PYTHON"] = ""
-    env["R_HOME"] = "*"
+    # 9. HOME. Lots of things need HOME.
 
-    cmd = Cmd(`$(Base.julia_cmd()) -e $(code)`;
-              env = env)
+    env = Dict(
+        "JULIA_DEPOT_PATH" => tmp_dir_2,
+        "JULIA_REGISTRYCI_AUTOMERGE" => "true",
+        "PYTHON" => "",
+        "R_HOME" => "*",
+    )
+    for k in ("HOME", "PATH", "HTTP_PROXY", "HTTPS_PROXY", "JULIA_PKG_SERVER")
+        if haskey(ENV, k)
+            env[k] = ENV[k]
+        end
+    end
+
+    cmd = Cmd(`$(Base.julia_cmd()) -e $(code)`; env=env)
+
     # GUI toolkits may need a display just to load the package
     xvfb = Sys.which("xvfb-run")
     @info("xvfb: ", xvfb)

